File: apps/android/core/model/src/main/kotlin/app/gauja/core/model/status/IssueStatus.kt
// SPDX-FileCopyrightText: 2026 Gauja contributors
// SPDX-License-Identifier: AGPL-3.0-or-later
package app.gauja.core.model.status

// Seerr server/constants/issue.ts at 69f73a6f1486fdb51b8ddae9a94a8dfb629f461c.
sealed class IssueStatus(val rawValue: Int?) {
    data object OPEN : IssueStatus(1)

    data object RESOLVED : IssueStatus(2)

    data class Unrecognized(val value: Int?) : IssueStatus(value)

    companion object {
        fun fromWire(raw: Int?): IssueStatus =
            when (raw) {
                1 -> OPEN
                2 -> RESOLVED
                else -> Unrecognized(raw)
            }
    }
}

File: apps/ios/Packages/Model/Sources/Model/status/IssueStatus.swift
// SPDX-FileCopyrightText: 2026 Gauja contributors
// SPDX-License-Identifier: AGPL-3.0-or-later
// Seerr server/constants/issue.ts at 69f73a6f1486fdb51b8ddae9a94a8dfb629f461c.
public enum IssueStatus: Sendable, Equatable, Codable {
    case open
    case resolved
    case unrecognized(Int?)

    public init(rawValue: Int?) {
        switch rawValue {
        case 1: self = .open
        case 2: self = .resolved
        default: self = .unrecognized(rawValue)
        }
    }

    public var rawValue: Int? {
        switch self {
        case .open: 1
        case .resolved: 2
        case .unrecognized(let raw): raw
        }
    }
}
